examples/cis2-wccd/src/lib.rs

```diff
@@ -1204,6 +1222,73 @@ mod tests {
         );
     }
 
+    /// Test only admin can setMetadataUrl
+    #[concordium_test]
+    #[cfg(feature = "crypto-primitives")]
+    fn test_set_metadata_url() {
+        let mut logger = TestLogger::init();
+        let mut state_builder = TestStateBuilder::new();
+        let state = State::new(&mut state_builder, ADMIN_ADDRESS);
+        let mut host = TestHost::new(state, state_builder);
+
+        // Set up the context
+        let mut ctx = TestReceiveContext::empty();
+        ctx.set_sender(ADMIN_ADDRESS);
+
+        // Set up crypto primitives to hash the document.
+        let crypto_primitives = TestCryptoPrimitives::new();
+
+        // Create a new_url and a new_hash
+        let new_url = "https://some.example/token/wccd/updated".to_string();
+        let new_hash = Some(crypto_primitives.hash_sha2_256("document".as_bytes()).0);
+
+        // Set up the parameter.
+        let parameter = SetMetadataUrlParams {
+            url:  new_url.clone(),
+            hash: new_hash,
```

Comment:
Nitpick, feel free to ignore:
```suggestion
        let new_hash = crypto_primitives.hash_sha2_256("document".as_bytes()).0;

        // Set up the parameter.
        let parameter = SetMetadataUrlParams {
            url:  new_url.clone(),
            hash: Some(new_hash),
```